zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +171,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
+        }
+        return subdomain;
+    }
+
+    private void checkOrchestratorZoneExists(String name) throws ZoneAlreadyExistsException {
+        IdentityZone identityZone = null;
+        try{
+            identityZone = zoneProvisioning.retrieveByName(name);
+        } catch (ZoneDoesNotExistsException e){
+            identityZone = null;
+        }
+        if(identityZone != null){
+            throw new ZoneAlreadyExistsException("Orchestrator zone already exists for name:  " + name );
+        }
+    }
+
+    private void createClient(String adminClientSecret, String subdomain, IdentityZone created)
+        throws OrchestratorZoneServiceException {
+        String zoneId = IdentityZoneHolder.get().getId();
+        String authorities = zoneAuthorities + ",zones." + zoneId + ".admin";
+        try {
+            createClient(created.getId(), authorities, clientId, adminClientSecret, grantTypes, resourceIds,
+                         scopes);
+        } catch (Exception e) {
+            logger.error("Unable to create client for subdomain : " + subdomain + " Zone Id : " + zoneId + " Exception is :" + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create client for " +
+                                                       "subdomain : " + subdomain + " Zone Id : " + zoneId + " Exception is :" + e.getMessage());
+        }
+    }
+
+    private void createDefaultIdp(IdentityZone created) throws OrchestratorZoneServiceException {
+        try {
+            IdentityProvider defaultIdp = new IdentityProvider();
+            defaultIdp.setName(OriginKeys.UAA);
+            defaultIdp.setType(OriginKeys.UAA);
+            defaultIdp.setOriginKey(OriginKeys.UAA);
+            defaultIdp.setIdentityZoneId(created.getId());
+            UaaIdentityProviderDefinition idpDefinition = new UaaIdentityProviderDefinition();
+            idpDefinition.setPasswordPolicy(null);
+            defaultIdp.setConfig(idpDefinition);
+            idpProvisioning.create(defaultIdp, created.getId());
+        } catch (Exception e) {
+            logger.error("Unable to create identity provider for subdomain : " + created.getSubdomain() + " Exception " +
+                         "is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create identity provider for " +
+                                                       "subdomain : ." + created.getSubdomain() + " Exception is : " + e.getMessage());
+        }
+    }
+
+    private IdentityZone createIdentityZone(IdentityZone identityZone)
+        throws OrchestratorZoneServiceException {
+        IdentityZone created = null;
+        try {
+            created = zoneProvisioning.create(identityZone);
+        } catch (Exception e) {
+            logger.error("Unable to create identity zone for subdomain : " + identityZone.getSubdomain() + " Exception is : " + e.getMessage());
```

Comment:
Log entire stack trace instead of printing only message. This will helpful for debug.

Reply:
Done. Thanks!